server/dash/pages/nc_summary_comparison.py

```diff
@@ -58,6 +58,133 @@ def merge_dict(d1, d2):
 COMPARISON_DASHBOARD_TITLE = "LA 311 Requests - Neighborhood Council Comparison Dashboard"
 
 
+## DATA WRANGLING HELPER FUNCTIONS.
+def add_datetime_column(df, colname):
+    """Adds a datetime column to a dataframe.
+    This function takes a datetime column 'colname' in string type, remove the last 4 characters,
+    split date and time by character 'T', and finally combine date and time into a single string again.
+    The function then converts the string using pandas's to_datetime function.
+    Args:
+        df: dataframe to add the datetime column to.
+        colname: the datetime column that is in string type.
+    
+    Return:
+        A dataframe with new column 'colnameDT' that is in datetime type.
+    """
+    df.loc[:, colname+"DT"] = pd.to_datetime(
+        df.loc[:, colname].str[:-4].str.split("T").str.join(" "))
+    return df
+
+def generate_filtered_dataframe(api_data_df, selected_nc, selected_request_types):
+    """Outputs the filtered dataframe based on the selected filters
+    This function takes the original dataframe "api_data_df", selected neighborhood
+    council "selected_nc", as well as the selected request type "selected_request_types" 
+    to output a dataframe with matching records.
+    Args:
+        api_data_df: full 311-request data directly access from the 311 data API.
+        selected_nc: A string argument automatically detected by Dash callback 
+        function when "selected_nc" element is selected in the layout.
+        selected_request_types: A list of strings automatically detected by Dash 
+        callback function when "selected_request_types" element is selected in the
+         layout, default None.
+    Returns:
+        pandas dataframe filtered by selected neighborhood council and request types
+    """
+    if (not selected_request_types or selected_request_types == ' ') and not selected_nc:
```

Comment:
let's create a var for the first part of this boolean statement since it's used twice

Reply:
thanks for the suggestion - resolved.